src/transformers/trainer.py

```diff
@@ -4083,15 +4083,32 @@ def _add_sm_patterns_to_gitignore(self) -> None:
             self.repo.git_push()
 
     def create_accelerator_and_postprocess(self):
-        grad_acc_kwargs = {"num_steps": self.args.gradient_accumulation_steps}
+        grad_acc_kwargs = {}
+        if self.args.accelerator_config.gradient_accumulation_kwargs is not None:
+            grad_acc_kwargs = self.args.accelerator_config.gradient_accumulation_kwargs
+
+        # check if num_steps is attempted to be passed in gradient_accumulation_kwargs
+        if "num_steps" in grad_acc_kwargs:
+            warnings.warn(
+                '"num_steps" passed in gradient_accumulation_kwargs will be ignored in favour of TrainingArguments.gradient_accumulation_steps.'
+            )
```

Comment:
Let's change this to instead check if the user passes in a custom plugin with n_steps > 1, and training args == 1. Let it passthrough as that means we're configuring via this, which is acceptable (we do this in deepspeed too)

Reply:
Otherwise raise a full explicit error, not a warning

Reply:
sounds good. I have implemented the logic that you described here. Also I have added a new `test_accelerator_config_from_dict_grad_accum_num_steps`  that checks the following 3 conditions:
1. TrainingArguments.gradient_accumulation_steps == 1 and  gradient_accumulation_kwargs['num_steps] == 1
   * this is a complete pass through with no warning
2. TrainingArguments.gradient_accumulation_steps == 1 and gradient_accumulation_kwargs['num_steps] > 1
    * this will pass through with a warning that now accum steps is taken from `gradient_accumulation_kwargs`
3. TrainingArguments.gradient_accumulation_steps > 1 and gradient_accumulation_kwargs['num_steps] was specified
    * this will raise an Exception

Reply:
resolving this. addressed in later comments.